evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -342,39 +444,32 @@ def get_trend_dataframe(self, X, y):
 
         """
         X = infer_feature_types(X)
-        if not isinstance(X.index, pd.DatetimeIndex):
-            raise TypeError("Provided X should have datetimes in the index.")
-        if X.index.freq is None:
-            raise ValueError(
-                "Provided DatetimeIndex of X should have an inferred frequency.",
-            )
+
         # Change the y index to a matching datetimeindex or else we get a failure
         # in ForecastingHorizon during decomposition.
         if not isinstance(y.index, pd.DatetimeIndex):
             y = self._set_time_index(X, y)
 
         self._check_oos_past(y)
 
-        result_dfs = []
-
-        def _decompose_target(X, y, fh):
+        def _decompose_target(X, y, fh, trend, seasonal, residual, period, id):
```

Comment:
I'm not sure I understand what you mean by this. If I just try to take out the variables it causes an error saying they're referenced before assignment.

Reply:
Which variable is throwing that error? It may require a renaming or two, but it should work

Reply:
Oh wait I think it is just because `_decompose_target` is defined before the variables since the variables are set inside the for loop. Would it be better to pass in the variables or take them out and define `_decompose_target` inside of the for loop?

Reply:
trend, seasonal and residual threw the error

Reply:
Ah, ok - I missed a bit of Python technicality here. Since we're reassigning `trend`, `seasonal`, and `residual` within the function, the Python interpreter treats them as local variables and doesn't consider them to necessarily be the same as those in the outer scope. It's fine to keep those as arguments that need passing in!